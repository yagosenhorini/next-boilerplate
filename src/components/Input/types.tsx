export interface IInput extends React.InputHTMLAttributes<HTMLInputElement> {
  value: string;
  onChange: () => void;
  placeholder: string;
}
