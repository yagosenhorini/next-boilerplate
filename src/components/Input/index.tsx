import * as S from './styled';
import { IInput } from './types';

const Input = ({ value, onChange, type, placeholder }: IInput) => (
  <S.Input
    type={type}
    value={value}
    onChange={onChange}
    placeholder={placeholder}
    data-testid="input-component"
  />
);

export default Input;
