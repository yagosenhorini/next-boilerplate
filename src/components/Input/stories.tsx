import { Meta, StoryObj } from '@storybook/react';
import Input from '.';

export default {
  title: 'Input',
  component: Input,
  tags: ['autodocs'],
} as Meta;

export const Default: StoryObj = {};
