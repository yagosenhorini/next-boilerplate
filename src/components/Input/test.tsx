import { render, screen } from '@testing-library/react';

import Input from '.';

import StyledProvider from '../../lib/theme';

describe('<Input />', () => {
  const mockValue = '';
  const mockOnChange = jest.fn();
  it('should render the input', () => {
    const { container } = render(
      <StyledProvider>
        <Input
          type="text"
          placeholder="text"
          value={mockValue}
          onChange={mockOnChange}
        />
      </StyledProvider>
    );

    expect(screen.getByTestId('input-component')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
