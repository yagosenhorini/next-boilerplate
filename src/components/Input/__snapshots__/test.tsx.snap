// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render the input 1`] = `
.c0 {
  width: 100%;
  border: none;
  height: 3.75rem;
  padding-left: 1.25rem;
  border-radius: 2.5rem;
  border: 1px solid #000;
}

.c0.has--error {
  border-color: #82090f;
}

.c0.is--filled {
  border-color: green;
}

<input
  class="c0"
  data-testid="input-component"
  placeholder="text"
  type="text"
  value=""
/>
`;
